### PR TITLE
yabai: update to 7.1.4

### PR DESCRIPTION
**Pull request by tsujp**
#### Description

update `yabai` to latest release.

###### Tested on

macOS 14.5 23F79 arm64
Command Line Tools 15.3.0.0.1.1708646388

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: sysutils/yabai/files/patch-makefile.diff [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+--- makefile	2024-09-30 00:05:50
++++ makefile.orig	2024-09-30 01:16:00
 @@ -1,7 +1,7 @@
  FRAMEWORK_PATH = -F/System/Library/PrivateFrameworks
  FRAMEWORK      = -framework Carbon -framework Cocoa -framework CoreServices -framework CoreVideo -framework SkyLight
  CLI_FLAGS      =
--BUILD_FLAGS    = -std=c99 -Wall -g -O0 -fvisibility=hidden -mmacosx-version-min=11.0 -fno-objc-arc -arch x86_64 -arch arm64 -sectcreate __TEXT __info_plist $(INFO_PLIST)
-+CFLAGS         += -std=c99 -Wall -DNDEBUG -O3 -fvisibility=hidden -fno-objc-arc -arch x86_64 -arch arm64 -sectcreate __TEXT __info_plist $(INFO_PLIST)
+-BUILD_FLAGS    = -std=c11 -Wall -Wextra -g -O0 -fvisibility=hidden -mmacosx-version-min=11.0 -fno-objc-arc -arch x86_64 -arch arm64 -sectcreate __TEXT __info_plist $(INFO_PLIST)
++CFLAGS         += -std=c11 -Wall -Wextra -DNDEBUG -O3 -fvisibility=hidden -fno-objc-arc -arch x86_64 -arch arm64 -sectcreate __TEXT __info_plist $(INFO_PLIST)
```
Comment (reneeotten) [inline sysutils/yabai/files/patch-makefile.diff:8]: if this now needs C11 then you should request a compatible compiler in the PortGroup. 

As with "compiler.c_standard 2011".

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: sysutils/yabai/files/patch-makefile.diff (hunk 1 above)
Comment (tsujp) [reply sysutils/yabai/files/patch-makefile.diff:8]: Upstream changed to c11.

Will enact requested change.

Comment (reneeotten) [reply sysutils/yabai/files/patch-makefile.diff:8]: ping - please address this so we can merge the PR

Comment (tsujp) [reply sysutils/yabai/files/patch-makefile.diff:8]: @reneeotten sorry, lots going on and I forgot. Updated.

**Pull request merged**